src/opm/input/eclipse/Schedule/KeywordHandlers.cpp

```diff
@@ -580,68 +583,38 @@ File {} line {}.)", wname, location.keyword, location.filename, location.lineno)
-                            if (!apply_default_gas_target)
-                                 production.production_controls += static_cast<int>(Group::ProductionCMode::GRAT);
-                            if (!apply_default_water_target)
-                                 production.production_controls += static_cast<int>(Group::ProductionCMode::WRAT);
-                        }
-                    }
-                    else {
-                        if (exceedAction == Group::ExceedAction::RATE) {
-                            if (!apply_default_oil_target)
-                                 production.production_controls += static_cast<int>(Group::ProductionCMode::ORAT);
-                            if (!apply_default_water_target)
-                                 production.production_controls += static_cast<int>(Group::ProductionCMode::WRAT);
-                            if (!apply_default_gas_target)
-                                 production.production_controls += static_cast<int>(Group::ProductionCMode::GRAT);
-                            if (!apply_default_liquid_target)
-                                 production.production_controls += static_cast<int>(Group::ProductionCMode::LRAT);
-                        }
-                    }
+                     if (production.cmode == Group::ProductionCMode::ORAT ||
+                         (exceedAction == Group::ExceedAction::RATE && !apply_default_oil_target)) {
+                         production.production_controls += static_cast<int>(Group::ProductionCMode::ORAT);
+                     }
+                     if (production.cmode == Group::ProductionCMode::WRAT ||
+                              ((exceedAction == Group::ExceedAction::RATE ||
+                                waterExceedAction == Group::ExceedAction::RATE)
+                                && !apply_default_water_target)) {
+                                production.production_controls += static_cast<int>(Group::ProductionCMode::WRAT);
+                     }
+                     if (production.cmode == Group::ProductionCMode::GRAT ||
+                              ((exceedAction == Group::ExceedAction::RATE ||
+                                gasExceedAction == Group::ExceedAction::RATE)
+                                && !apply_default_gas_target)) {
+                                production.production_controls += static_cast<int>(Group::ProductionCMode::GRAT);
+                     }
+                     if (production.cmode == Group::ProductionCMode::LRAT ||
+                              ((exceedAction == Group::ExceedAction::RATE ||
+                                liquidExceedAction == Group::ExceedAction::RATE)
+                                && !apply_default_liquid_target)) {
+                                production.production_controls += static_cast<int>(Group::ProductionCMode::LRAT);
+                     }
```

Comment:
Would you mind using a bitwise "or", i.e.,
```C++
production.production_controls |= static_cast<int>(...);
```
here please?  That's a safer operation for this particular application.